### ISSUE TITLE
Error searching file: Got response (400)

### ISSUE THREAD

**Issue by delvin-fil** [portagefilelist/client#21]
Error searching file: Got response (400): {"error":{"code":"SEARCH_FAILED","message":"Search resulted in an unknow error."}}
Traceback (most recent call last):
  File "/usr/lib/python-exec/python3.10/e-file", line 36, in <module>
    if len(files['result']) > 0:
KeyError: 'result'

**Comment by bananas-repos** [comment]
Hello.

Can you please provide the complete command you have used?

There was an error on the endpoint and it should be working again.

**Comment by delvin-fil** [comment]
```
e-file pulseaudio-equalizer-gtk
Error searching file: Got response (400): {"error":{"code":"SEARCH_FAILED","message":"Search resulted in an unknow error."}}
Traceback (most recent call last):
  File "/usr/lib/python-exec/python3.10/e-file", line 36, in <module>
    if len(files['result']) > 0:
KeyError: 'result'
```
```
e-file rst2man
Error searching file: Got response (400): {"error":{"code":"SEARCH_FAILED","message":"Search resulted in an unknow error."}}
Traceback (most recent call last):
  File "/usr/lib/python-exec/python3.10/e-file", line 36, in <module>
    if len(files['result']) > 0:
KeyError: 'result'
```
```
e-file -v
e-file 3.1
```
```
eix app-portage/pfl
[I] app-portage/pfl
     Доступные версии:      3.2-r2 {+network-cron PYTHON_TARGETS="python3_10 python3_11"}
     Установленные версии:  3.2-r2(06:47:35 08.07.2023)(network-cron PYTHON_TARGETS="python3_10 python3_11")
     Домашняя страница:     http://www.portagefilelist.de https://github.com/portagefilelist/client
     Описание:              Searchable online file/package database for Gentoo
```

**Comment by delvin-fil** [comment]
The error is caused by something unclear. At first, I sinned on python-exec, because when called directly, everything worked (before entering, for example, "rst2man"). Now I don't know what to think.
Here: https://pastebin.com/utn8qQEU

**Comment by bananas-repos** [comment]
Thank you the feedback!

> e-file pulseaudio-equalizer-gtk
Currently this is not known yet

> e-file rst2man
Is this the real filename? Currently there is file with the name rest2man.py

The messages from the endpoint and in the e-file itself are not very good. It is on my todo list to make it better. The user does not really get the correct feedback.

The next problem is, that the search needs either the exact filename or the extact path and filename

```
e-file /usr/bin/rst2man.py
[I] dev-python/docutils
        Seen Versions:          0.20.1
        Portage Versions:       0.17.1-r1 0.19 0.20.1 
        Installed Versions:     0.20.1(Wed Jun 28 08:53:35 2023) 
        Homepage:               https://docutils.sourceforge.io/ https://pypi.org/project/docutils/
        Description:            Python Documentation Utilities (reference reStructuredText impl.)
        Matched Files:          /usr/bin/rst2man.py/rst2man.py`
```

```
e-file rst2man.py
[I] dev-python/docutils
        Seen Versions:          0.20.1
        Portage Versions:       0.17.1-r1 0.19 0.20.1 
        Installed Versions:     0.20.1(Wed Jun 28 08:53:35 2023) 
        Homepage:               https://docutils.sourceforge.io/ https://pypi.org/project/docutils/
        Description:            Python Documentation Utilities (reference reStructuredText impl.)
        Matched Files:          /usr/bin/rst2man.py/rst2man.py; /usr/lib/python-exec/pypy3/rst2man.py/rst2man.py; /usr/lib/python-exec/python3.10/rst2man.py/rst2man.py; /usr/lib/python-exec/python3.11/rst2man.py/rst2man.py; /usr/lib/python-exec/python3.12/rst2man.py/rst2man.py; /usr/share/doc/docutils-0.20.1/html/tools/rst2man.py/rst2man.py
```

or use a wildcard

```
e-file rst2man*
[I] dev-python/docutils
        Seen Versions:          0.20.1
        Portage Versions:       0.17.1-r1 0.19 0.20.1 
        Installed Versions:     0.20.1(Wed Jun 28 08:53:35 2023) 
        Homepage:               https://docutils.sourceforge.io/ https://pypi.org/project/docutils/
        Description:            Python Documentation Utilities (reference reStructuredText impl.)
        Matched Files:          /usr/bin/rst2man.py/rst2man.py; /usr/lib/python-exec/pypy3/rst2man.py/rst2man.py; /usr/lib/python-exec/python3.10/rst2man.py/rst2man.py; /usr/lib/python-exec/python3.11/rst2man.py/rst2man.py; /usr/lib/python-exec/python3.12/rst2man.py/rst2man.py; /usr/share/doc/docutils-0.20.1/html/tools/rst2man.py/rst2man.py
```

So I need to improve the e-file, the endpoint and the user documentation.

**Comment by delvin-fil** [comment]
I figured out rst2man myself, thank you.
But it doesn't work with pulseaudio-equalizer-gtk.
It exists: https://www.google.com/search?q=gentoo+%22pulseaudio-equalizer-gtk%22

**Comment by bananas-repos** [comment]
I'm not really sure if a package or either a file with the name "pulseaudio-equalizer-gtk" really exists. My google results do not show a valid result which tells me there is.

portagefilelist does only know files and packages from users who commit their installed packages and files. It does not know the whole files from gentoo.

**Comment by bananas-repos** [comment]
Will be fixed in release 3.2.1

**Comment by delvin-fil** [comment]
thank you.
my turnips are not updated.
I will update according to the schedule - on the last night
in this case - from the 29th to the 30th
